src/InMemoryRepository.php

```diff
@@ -256,11 +269,55 @@ private function removeResource(PuliResource $resource)
 
         unset($this->resources[$path]);
 
+        if ($resource instanceof LinkResource && array_key_exists($path, $this->links)) {
+            unset($this->links[$path]);
+        }
+
         // Detach from locator
         $resource->detach();
     }
 
     /**
+     * {@inheritdoc}
+     */
+    protected function followLinks($path)
+    {
+        foreach ($this->links as $link) {
+            $linkPath = rtrim($link->getPath(), '/');
+
+            if (0 === strpos($path, $linkPath.'/')) {
+                $realTargetPath = rtrim($this->getRealTarget($link)->getPath(), '/');
+                $realPath = substr_replace($path, $realTargetPath, 0, strlen($linkPath));
+
+                if (array_key_exists($realPath, $this->resources)) {
+                    return $this->resources[$realPath];
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Find the real target of a given link (resolve all the links until
+     * a non-link resource is found).
+     *
+     * @param LinkResource $linkResource
+     *
+     * @return PuliResource
+     */
+    private function getRealTarget(LinkResource $linkResource)
```

Comment:
You could add this logic to `LinkResource::getTarget()` and activate it through a `$followLinks` argument (`false` by default).